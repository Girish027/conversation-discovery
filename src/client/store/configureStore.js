import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { createLogger } from 'redux-logger';
import thunk from 'redux-thunk';
import { routerMiddleware } from 'connected-react-router/immutable';
import {
  fromJS,
} from 'immutable';
import createRootReducer from './reducers';

const loggerMiddleware = createLogger({
  predicate: () => process.env.NODE_ENV === 'development',
  stateTransformer: (state) => JSON.parse(JSON.stringify(state)),
  actionTransformer: (action) => JSON.parse(JSON.stringify(action)),
});


const configureStore = (initialState = {}, history) => {
  const middlewares = [
    routerMiddleware(history), // for dispatching history actions
    thunk,
    loggerMiddleware
  ];

  const store = createStore(
    createRootReducer(history),
    fromJS(initialState),
    composeWithDevTools(applyMiddleware(...middlewares))
  );

  // Hot reloading
  if (module.hot) {
    // Enable Webpack hot module replacement for reducers
    module.hot.accept('./reducers', () => {
      store.replaceReducer(createRootReducer(history));
    });
  }

  return store;
};

export default configureStore;
