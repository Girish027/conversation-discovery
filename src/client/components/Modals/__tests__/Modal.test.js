import React from 'react';
import { mount, shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ConnectedModal, { Modal } from 'components/Modals/Modal';
import { fromJS } from 'immutable';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Constants from 'components/../constants';

describe('<Modal />', function () {
  let wrapper;
  let store;

  const caa = {
    clientId: '247inc',
    appId: 'referencebot',
    accountId: 'referencebot',
  };

  const modalState = {
    modalName: Constants.modals.createProject,
    header: 'Create a new project',
    closeIconVisible: true,
  };

  const props = {
    caa,
    modalState,
    dispatch: () => {}
  };

  beforeAll(() => {
    // Mock store
    const mockStore = configureStore();
    store = mockStore(fromJS({
      app: {
        ...caa,
        modalState
      },
      projects: {
        projectCreationInProgress: false
      },
      conversations: {
        conversationsAddingInProgress: false
      }
    }));
  });


  afterAll(() => {
    jest.clearAllMocks();
  });

  const getConnectedWrapper = () => mount(
    <Provider store={store}>
      <ConnectedModal />
    </Provider>);

  const getShallowWrapper = (propsObj) => shallow(<Modal
    {...propsObj}
  />);

  describe('Creating an instance', () => {
    test('should exist - connected component Modal', () => {
      wrapper = getConnectedWrapper();
      expect(wrapper.exists()).toBe(true);
    });

    test('should exist - basic Modal component', () => {
      wrapper = getShallowWrapper(props);
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Snapshots', () => {
    test('renders correctly with CreateProject Modal', () => {
      wrapper = shallow(<Modal
        {...props}
      />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly with Confirmation Modal', () => {
      wrapper = shallow(<Modal
        {...props}
        modalState={{
          modalName: Constants.modals.confirm,
          header: 'Confirm Action',
          message: 'Please confirm the action'
        }}
      />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly with Progress Modal', () => {
      wrapper = shallow(<Modal
        {...props}
        modalState={{
          modalName: Constants.modals.progress,
        }}
      />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly with AddToBot Modal', () => {
      wrapper = shallow(<Modal
        {...props}
        modalState={{
          modalName: Constants.modals.addToBot,
        }}
      />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly with AddToFaq Modal', () => {
      wrapper = shallow(<Modal
        {...props}
        modalState={{
          modalName: Constants.modals.addToFaq,
        }}
      />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly with Simple Modal for unauthorized user', () => {
      wrapper = shallow(<Modal
        {...props}
        modalState={{
          modalName: Constants.modals.unauthorized,
          message: 'you are unauthorized to access this client',
          header: 'Access Denied',
        }}
      />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders null when modalState is empty', () => {
      wrapper = shallow(<Modal
        {...props}
        modalState={{}}
      />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe('Functionality:', () => {
    beforeEach(() => {
      global.window.document.addEventListener = jest.fn();
      global.window.document.removeEventListener = jest.fn();
    });

    afterEach(() => {
      jest.clearAllMocks();
      store.clearActions();
    });

    describe('componentDidMount', () => {
      test('should add event listener for keydown event', () => {
        wrapper = getShallowWrapper(props);
        wrapper.instance().componentDidMount();
        expect(global.window.document.addEventListener).toHaveBeenCalledWith('keydown', wrapper.instance().handleEscapeKey, false);
      });
    });

    describe('componentWillUnmount', () => {
      test('should remove event listener for keydown event', () => {
        wrapper = getShallowWrapper(props);
        wrapper.instance().componentWillUnmount();
        expect(global.window.document.removeEventListener).toHaveBeenCalledWith('keydown', wrapper.instance().handleEscapeKey, false);
      });
    });

    describe('closeModal', () => {
      test('should dispatch action to close modal', () => {
        wrapper = getConnectedWrapper().find(Modal);
        wrapper.instance().closeModal();
        expect(store.getActions()).toMatchSnapshot();
      });
    });

    describe('handleEscapeKey', () => {
      test('should dispatch action to close modal when escape key is pressed', () => {
        wrapper = getConnectedWrapper().find(Modal);
        wrapper.instance().handleEscapeKey({
          keyCode: 27,
          preventDefault: () => {}
        });
        expect(store.getActions()).toMatchSnapshot();
      });

      test('should not dispatch action to close modal when other keys are pressed', () => {
        wrapper = getConnectedWrapper().find(Modal);
        wrapper.instance().handleEscapeKey({
          keyCode: 13,
          preventDefault: () => {}
        });
        expect(store.getActions()).toEqual([]);
      });
    });
  });
});
