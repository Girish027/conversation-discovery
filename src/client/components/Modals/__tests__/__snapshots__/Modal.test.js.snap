// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> Functionality: closeModal should dispatch action to close modal 1`] = `
Array [
  Object {
    "modalIsOpen": false,
    "modalState": Object {},
    "type": "app/MODAL_IS_OPEN",
  },
  Object {
    "isComplete": true,
    "type": "conversations/ADD_CONVERSATIONS_COMPLETE",
  },
]
`;

exports[`<Modal /> Functionality: handleEscapeKey should dispatch action to close modal when escape key is pressed 1`] = `
Array [
  Object {
    "modalIsOpen": false,
    "modalState": Object {},
    "type": "app/MODAL_IS_OPEN",
  },
]
`;

exports[`<Modal /> Snapshots renders correctly with AddToBot Modal 1`] = `
<Connect(AddToBotModal)
  conversationsAddingInProgress={false}
  dispatch={[Function]}
  onClickCancel={[Function]}
  onClickClose={[Function]}
  styleOverride={
    Object {
      "cancel": Object {
        "backgroundColor": "unset",
        "paddingLeft": "10px",
        "paddingRight": "10px",
      },
      "ok": Object {
        "marginLeft": "10px",
        "paddingLeft": "25px",
        "paddingRight": "25px",
      },
      "top": "20%",
    }
  }
/>
`;

exports[`<Modal /> Snapshots renders correctly with AddToFaq Modal 1`] = `
<Connect(AddToFaqModal)
  conversationsAddingInProgress={false}
  dispatch={[Function]}
  onClickCancel={[Function]}
  onClickClose={[Function]}
  styleOverride={
    Object {
      "cancel": Object {
        "backgroundColor": "unset",
        "paddingLeft": "10px",
        "paddingRight": "10px",
      },
      "ok": Object {
        "marginLeft": "10px",
        "paddingLeft": "25px",
        "paddingRight": "25px",
      },
      "top": "20%",
    }
  }
/>
`;

exports[`<Modal /> Snapshots renders correctly with Confirmation Modal 1`] = `
<ConfirmationModal
  caa={
    Object {
      "accountId": "referencebot",
      "appId": "referencebot",
      "clientId": "247inc",
    }
  }
  closeIconVisible={false}
  confirmMode={false}
  deleteMode={false}
  dispatch={[Function]}
  header="Confirm Action"
  message="Please confirm the action"
  onClickCancel={[Function]}
  onClickClose={[Function]}
  onClickOk={[Function]}
  styleOverride={
    Object {
      "cancel": Object {
        "backgroundColor": "unset",
        "paddingLeft": "10px",
        "paddingRight": "10px",
      },
      "ok": Object {
        "marginLeft": "10px",
        "paddingLeft": "25px",
        "paddingRight": "25px",
      },
      "top": "20%",
    }
  }
/>
`;

exports[`<Modal /> Snapshots renders correctly with CreateProject Modal 1`] = `
<CreateProjectModal
  caa={
    Object {
      "accountId": "referencebot",
      "appId": "referencebot",
      "clientId": "247inc",
    }
  }
  closeIconVisible={true}
  dispatch={[Function]}
  error=""
  errorType=""
  formdata={Object {}}
  header="Create a new project"
  onClickCancel={[Function]}
  onClickClose={[Function]}
  projectCreationInProgress={false}
  styleOverride={
    Object {
      "cancel": Object {
        "backgroundColor": "unset",
        "paddingLeft": "10px",
        "paddingRight": "10px",
      },
      "ok": Object {
        "marginLeft": "10px",
        "paddingLeft": "25px",
        "paddingRight": "25px",
      },
      "top": "20%",
    }
  }
/>
`;

exports[`<Modal /> Snapshots renders correctly with Progress Modal 1`] = `
<ProgressModal
  caa={
    Object {
      "accountId": "referencebot",
      "appId": "referencebot",
      "clientId": "247inc",
    }
  }
  closeIconVisible={false}
  dispatch={[Function]}
  message=""
  onClickCancel={[Function]}
  onClickCancelButton={[Function]}
  onClickClose={[Function]}
  showCancelButton={false}
  styleOverride={
    Object {
      "cancel": Object {
        "backgroundColor": "unset",
        "paddingLeft": "10px",
        "paddingRight": "10px",
      },
      "ok": Object {
        "marginLeft": "10px",
        "paddingLeft": "25px",
        "paddingRight": "25px",
      },
      "top": "20%",
    }
  }
/>
`;

exports[`<Modal /> Snapshots renders correctly with Simple Modal for unauthorized user 1`] = `
<SimpleModal
  closeIconVisible={true}
  dispatch={[Function]}
  header="Access Denied"
  message="you are unauthorized to access this client"
  onClickCancel={[Function]}
  onClickClose={[Function]}
  onClickOkButton={[Function]}
  styleOverride={
    Object {
      "cancel": Object {
        "backgroundColor": "unset",
        "paddingLeft": "10px",
        "paddingRight": "10px",
      },
      "ok": Object {
        "marginLeft": "10px",
        "paddingLeft": "25px",
        "paddingRight": "25px",
      },
      "top": "20%",
    }
  }
/>
`;

exports[`<Modal /> Snapshots renders null when modalState is empty 1`] = `""`;
